skimage/measure/_moments.py

```diff
@@ -1,11 +1,137 @@
+     * Centroid as: {``M[1, 0] / M[0, 0]``, ``M[0, 1] / M[0, 0]``}.
+
+    Note that raw moments are neither translation, scale nor rotation
+    invariant.
+
+    Parameters
+    ----------
+    contour : (N, D) double or uint8 array
+        Array of N points that describe an image of D dimensionality in
+        Euclidian space.
+    order : int, optional
+        Maximum order of moments. Default is 3.
+
+    Returns
+    -------
+    m : (``order + 1``, ``order + 1``) array
+        Raw image moments.
+
+    References
+    ----------
+    .. [1] Johannes Kilian. Simple Image Analysis By Moments. Durham
+           University, version 0.2, Durham, 2001.
+
+    Examples
+    --------
+    >>> contour = np.array([[r, c] for r in range(13, 17) \
+                  for c in range(13, 17)], dtype=np.double)
+    >>> M = moments_contour(contour)
+    >>> cr = M[1, 0] / M[0, 0]
+    >>> cc = M[0, 1] / M[0, 0]
+    >>> cr, cc
+    (14.5, 14.5)
+    """
+    return moments_contour_central(contour, 0, order=order)
+
+
+def moments_contour_central(contour, center=None, order=3):
+    """Calculate all raw image moments up to a certain order.
+
+    The following properties can be calculated from raw image moments:
+     * Area as: ``M[0, 0]``.
+     * Centroid as: {``M[1, 0] / M[0, 0]``, ``M[0, 1] / M[0, 0]``}.
+
+    Note that raw moments are neither translation, scale nor rotation
+    invariant.
+
+    Parameters
+    ----------
+    contour : (N, D) double or uint8 array
+        Array of N points that describe an image of D dimensionality in
+        Euclidian space.
+    center : tuple of float, optional
+        Coordinates of the image centroid. This will be computed if it
+        is not provided.
+    order : int, optional
+        Maximum order of moments. Default is 3.
+
+    Returns
+    -------
+    m : (``order + 1``, ``order + 1``) array
+        Raw image moments.
+
+    References
+    ----------
+    .. [1] Johannes Kilian. Simple Image Analysis By Moments. Durham
+           University, version 0.2, Durham, 2001.
+
+    Examples
+    --------
+    >>> contour = np.array([[r, c] for r in range(13, 17) \
+                  for c in range(13, 17)], dtype=np.double)
+    >>> M = moments_contour(contour)
+    >>> cr = M[1, 0] / M[0, 0]
+    >>> cc = M[0, 1] / M[0, 0]
+    >>> moments_contour_central(contour, (cr, cc))
+    array([[ 16.,   0.,  20.,   0.],
+           [  0.,   0.,   0.,   0.],
+           [ 20.,   0.,  25.,   0.],
+           [  0.,   0.,   0.,   0.]])
+    """
+    if isinstance(contour, tuple):
+        # Represent as array and rotate to standardize orientation.
+        contour = np.rollaxis(np.asarray(contour), axis=1)
+    assert_nD(contour, 2)
+    ndim = contour.shape[1]
+    if center is None:
+        M = moments_central(contour, center=0, order=order)
+        center = M[tuple(np.eye(ndim, dtype=int))]
+    else:
+        # TODO: Handle broadcasting of `center` argument.
+        if center is int:
```

Comment:
This is probably not doing what you think it's doing:

```python
In [1]: x = 5

In [2]: x is int
Out[2]: False

In [3]: y = int

In [4]: y is int
Out[4]: True
```

Reply:
And at any rate, if we want to support scalar centers (which I'm not sure we do), we should allow floats, also.

The function you're after is the builtin `isinstance` or the NumPy function `np.isscalar`.

Reply:
Actually I just saw what you're doing with `center` below. That will work without modification for scalar centers.

Reply:
Good catch – also ideally this would implement #2747 so that all similar nD functions that want to "broadcast" certain parameters, there is a single point of control.